moleculemaker/mmli-backend
# Added Helm chart + Dockerfile + Compose

## Problem
We will need a way to deploy this service to the cluster

## Approach
* Copy Docker + Compose + Helm chart from DMM, add minio
* Seal new secrets for staging service, and can add prod secrets once we're ready to deploy
* Includes (untested) configuration for backups for postgresql (minio tbd)

## How to Test
Prerequisites:
* local cluster running
* Install NGINX ingress: `helm upgrade --install ingress-nginx ingress-nginx   --repo https://kubernetes.github.io/ingress-nginx  --set controller.hostPort.enabled=true --namespace ingress-nginx --create-namespace `

Test Steps:
1. Install the helm chart on your local cluster: `helm upgrade --install mmli-backend -n mmli-backend . --create-namespace -f values.local.yaml`
```bash
Release "mmli-backend" has been upgraded. Happy Helming!
NAME: mmli-backend
LAST DEPLOYED: Thu Jun 15 11:01:17 2023
NAMESPACE: mmli-backend
STATUS: deployed
REVISION: 2
TEST SUITE: None
NOTES:
Application is now running!

Access http://mmli.fastapi.localhost to access FastAPI
```

2. Navigate your browser to http://mmli.fastapi.localhost to see Swagger UI
